### PR TITLE
New dashboard to monitor SMS send rate

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Add a new cloudwatch dashboard to monitor SMS sending (sms / fragments / send time / throttled vs pinpoint)

[ClickOps version on production](https://ca-central-1.console.aws.amazon.com/cloudwatch/home?region=ca-central-1#dashboards/dashboard/SJA-Specialized-monitor-SMS-send-rate)

image

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/428

## Before merging this PR

Read code suggestions left by the
[cds-ai-codereviewer](https://github.com/cds-snc/cds-ai-codereviewer/) bot. Address
valid suggestions and shortly write down reasons to not address others. To help
with the classification of the comments, please use these reactions on each of the
comments made by the AI review:

| Classification      | Reaction | Emoticon |
|---------------------|----------|----------|
| Useful              | +1       | 👍        |
| Noisy               | eyes     | 👀        |
| Hallucination       | confused | 😕        |
| Wrong but teachable | rocket   | 🚀        |
| Wrong and incorrect | -1       | 👎        |

The classifications will be extracted and summarized into an analysis of how helpful
or not the AI code review really is.

## Test instructions | Instructions pour tester la modification

_TODO: Fill in test instructions for the reviewer._

## Release Instructions | Instructions pour le déploiement

None.

## Reviewer checklist | Liste de vérification du réviseur

* [x] This PR does not break existing functionality.
* [x] This PR does not violate GCNotify's privacy policies.
* [x] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
* [x] This PR does not significantly alter performance.
* [x] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: aws/pinpoint_to_sqs_sms_callbacks/dashboards.tf [hunk 1]
```diff
@@ -70,7 +70,7 @@ resource "aws_cloudwatch_dashboard" "pinpoint" {
             "type": "metric",
             "properties": {
                 "metrics": [
-                    [ "AWS/SQS", "ApproximateAgeOfOldestMessage", "QueueName", "eks-notification-canada-cadelivery-receipts" ]
+                    [ "AWS/SQS", "ApproximateAgeOfOldestMessage", "QueueName", "${var.celery_queue_prefix}delivery-receipts" ]
```
Comment (sastels) [inline aws/pinpoint_to_sqs_sms_callbacks/dashboards.tf:73]: This dashboard didn't actually change, but there were some hardcoded names that could be replaced by terraform variables.

File: aws/pinpoint_to_sqs_sms_callbacks/dashboards.tf [hunk 2]
```diff
@@ -487,18 +487,145 @@
...
+        },
+        {
+            "height": 6,
+            "width": 6,
+            "y": 6,
+            "x": 18,
+            "type": "metric",
+            "properties": {
+                "metrics": [
+                    [ "ContainerInsights/Prometheus", "kube_deployment_status_replicas_available", "namespace", "notification-canada-ca", "ClusterName", "notification-canada-ca-production-eks-cluster", "deployment", "celery-sms-send-primary", { "region": "${var.region}", "label": "celery-sms-send-primary" } ],
```
[the remark this answers is not in the document]
Comment (sastels) [reply aws/pinpoint_to_sqs_sms_callbacks/dashboards.tf:554]: not a bad suggestion but we these are things we hard code everywhere.

**Review by ben851 — APPROVED** [review]
I think so

File: aws/pinpoint_to_sqs_sms_callbacks/dashboards.tf [hunk 3]
```diff
@@ -487,18 +487,145 @@ resource "aws_cloudwatch_dashboard" "pinpoint" {
...
             }
         }
     ]
 }
 EOF
 }
 
+resource "aws_cloudwatch_dashboard" "sms-send-rate" {
+  count          = var.cloudwatch_enabled ? 1 : 0
+  dashboard_name = "Specialized-sms-send-rate"
```
Comment (jimleroyer) [inline aws/pinpoint_to_sqs_sms_callbacks/dashboards.tf:507]: We could have English like names for the dashboard, just like the AWS provided ones. 

image

**Review by jimleroyer — APPROVED** [review]

**Pull request merged**